akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -680,7 +680,7 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
    */
   def leaving(address: Address): Unit = {
     // only try to update if the node is available (in the member ring)
-    if (latestGossip.members.exists(m ⇒ m.address == address && m.status == Up)) {
+    if (latestGossip.members.exists(m ⇒ m.address == address && (m.status == Joining || m.status == WeaklyUp || m.status == Up))) {
```

Comment:
I was pondering if this could have any unforeseen effects when merging membership states, which is done with `Member.highestPriorityOf`. As far as I can see it should be fine to change to Leaving. A later merge will keep the Leaving, and the leader will not change to Exiting until all have seen Leaving.